proof_of_concept/components/asset_store.py

```diff
@@ -14,17 +18,29 @@
 
 class AssetStore(IAssetStore):
     """A simple store for assets."""
-    def __init__(self, policy_evaluator: PolicyEvaluator) -> None:
+    def __init__(
+            self, policy_evaluator: PolicyEvaluator,
+            image_dir: Optional[Path] = None) -> None:
         """Create a new empty AssetStore."""
         self._policy_evaluator = policy_evaluator
         self._permission_calculator = PermissionCalculator(policy_evaluator)
         self._assets = dict()  # type: Dict[Identifier, Asset]
+        if image_dir is None:
+            image_dir = Path(mkdtemp())
+        self._image_dir = image_dir
 
-    def store(self, asset: Asset) -> None:
+    def close(self) -> None:
+        """Releases resources, call when done."""
+        rmtree(self._image_dir, ignore_errors=True)
+
+    def store(self, asset: Asset, move_image: bool = False) -> None:
```

Comment:
AssetStore.store sounds a bit funny, but i could not suggest any better alternative ('put' is also meh).

Reply:
Hmm, yes, English likes to verb its nouns... Maybe I could rename AssetStore to AssetWarehouse, AssetDepot, AssetBank, AssetDatabase or AssetMuseum (for curated data only!)? Although I guess for AssetDepot I'd have to go with AssetDepot.deposit(), so that may not solve it either :-).